query/resource_selector.go

```diff
@@ -224,24 +224,71 @@ func SetResourceSelectorClause(ctx context.Context, resourceSelector types.Resou
 		}
 	}
 
+	supportedFuncs := []string{"component_config_traverse"}
+	useFuncParsing := false
 	if resourceSelector.Search != "" {
-		var prefixQueries []*gorm.DB
-		if resourceSelector.Name == "" {
-			prefixQueries = append(prefixQueries, ctx.DB().Where("name ILIKE ?", resourceSelector.Search+"%"))
-		}
-		if resourceSelector.TagSelector == "" && table == "config_items" {
-			prefixQueries = append(prefixQueries, ctx.DB().Where("EXISTS (SELECT 1 FROM jsonb_each_text(tags) WHERE value ILIKE ?)", resourceSelector.Search+"%"))
+		for _, sf := range supportedFuncs {
+			if strings.Contains(resourceSelector.Search, sf) {
+				useFuncParsing = true
+				break
+			}
 		}
-		if resourceSelector.LabelSelector == "" {
-			prefixQueries = append(prefixQueries, ctx.DB().Where("EXISTS (SELECT 1 FROM jsonb_each_text(labels) WHERE value ILIKE ?)", resourceSelector.Search+"%"))
+		if useFuncParsing {
+			setSearchQueryParams(&resourceSelector)
+		} else {
+			var prefixQueries []*gorm.DB
+			if resourceSelector.Name == "" {
+				prefixQueries = append(prefixQueries, ctx.DB().Where("name ILIKE ?", resourceSelector.Search+"%"))
+			}
+			if resourceSelector.TagSelector == "" && table == "config_items" {
+				prefixQueries = append(prefixQueries, ctx.DB().Where("EXISTS (SELECT 1 FROM jsonb_each_text(tags) WHERE value ILIKE ?)", resourceSelector.Search+"%"))
+			}
+			if resourceSelector.LabelSelector == "" {
+				prefixQueries = append(prefixQueries, ctx.DB().Where("EXISTS (SELECT 1 FROM jsonb_each_text(labels) WHERE value ILIKE ?)", resourceSelector.Search+"%"))
+			}
+
+			query = OrQueries(query, prefixQueries...)
 		}
+	}
 
-		query = OrQueries(query, prefixQueries...)
+	if resourceSelector.Functions.ComponentConfigTraversal != nil {
+		args := resourceSelector.Functions.ComponentConfigTraversal
+		if table == "components" {
+			query = query.Where("id IN (SELECT id from lookup_component_config_id_related_components(?, ?, ?))", args.ComponentID, args.Direction, args.Types)
+		}
 	}
 
 	return query, nil
 }
 
+func setSearchQueryParams(rs *types.ResourceSelector) {
+	if rs.Search == "" {
+		return
+	}
+
+	queries := strings.Split(rs.Search, ";")
+	for _, q := range queries {
+		items := strings.Split(q, "=")
+		if len(items) != 2 {
+			continue
+		}
+
+		switch items[0] {
+		case "component_config_traverse":
```

Comment:
Can we add in the other fields that are currently in the cli catalog query ?

Reply:
Added